confdb/aclocal_cc.m4

```diff
@@ -1651,3 +1651,31 @@ AC_DEFUN([PAC_C_STATIC_ASSERT], [
     AC_MSG_RESULT([no])
     ])
 ])
+
+dnl
+dnl PAC_CC_CHECK_TLS - Test for thread local storage support
+dnl
+dnl will AC_DEFINE([TLS]) to a compiler supported TLS keyword
+dnl
+AC_DEFUN([PAC_CC_CHECK_TLS], [
+    AC_MSG_CHECKING([for thread-local storage])
+    if test -z $pac_cv_tls ; then
+        AC_LINK_IFELSE([AC_LANG_PROGRAM([_Thread_local int foo=0;],[foo=1])],
+            [pac_cv_tls=_Thread_local])
+    fi
+    if test -z $pac_cv_tls ; then
+        AC_LINK_IFELSE( [AC_LANG_PROGRAM([__thread int foo=0;],[foo=1])],
+            [pac_cv_tls=__thread])
+    fi
+    if test -z $pac_cv_tls ; then
+        AC_COMPILE_IFELSE( [AC_LANG_PROGRAM([__declspec(thread) int foo=0;],[foo=1])],
+            [pac_cv_tls="__declspec(thread)"])
```

Comment:
Is there a reason to use `AC_COMPILE_IFELSE` here? Actually, I am curious about which toolchain had problems with linking and wonder what causes it.

Reply:
This was a mistake. I am fixing it. The problematic toolchain is `icc` on macOS.